doc/en/developer/source/cite-test-guide/index.rst

```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+CITE automation tests with docker
+=================================
+
+
+How to automate the CITE Tests with
+`docker <https://www.docker.com>`_.
+
+Requirements:
+-------------
+
+- Running the tests requires a linux system with `docker <https://www.docker.com>`_, `docker-compose <https://docs.docker.com/compose/install>`_, and git installed on it.
+
 .. note::
 
-   To build every suite test you need to do the next steps:
+   The CITE tools are available in the build/cite folder of the `GeoServer repository <https://github.com/geoserver/geoserver/tree/master/build/cite>`_:
 
-  .. code:: shell
+Steps:
+------
 
-      git clone https://github.com/opengeospatial/ets-wcs10.git
-      cd ets-wcs10.git
-      mvn install
+**Set-up the environment.**
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Run WFS 1.0 tests
------------------
+   #.  Clone the repository.
 
-.. important::
+       .. code:: shell
+
+          git clone https://github.com/randomorder/geoserver.git --branch GSIP-176
+
+   #.  go the cite directory.
+
+       .. code:: shell
+
+          cd geoserver/build/cite
+
+   #.  inside will find a structure like below with a list of directories with the name of the suites to run.
+
+       .. code:: shell
+
+          cite
+          |-- forms
+          |-- geoserver
+          |-- run-test.sh
+          |-- wcs10
+          |-- wcs11
+          |-- wfs10
+          |-- wms11
+          |-- wms13
+          |-- wfs11
+          |-- logs
+          |-- docker-compose.yml
+          |-- postgres
+          |-- README.md
+          `-- Makefile
+
+**Running the suite tests.**
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   There is 2 way to run the suites, one is running with make that will
+   automate all the commands, and the second one is running the test through WebUI:
+
+   1. Running the through Makefile:
+
+      -  run ``make`` in the console, will give you the list of commands
+         to run.
+
+         .. code:: shell
+
+            make
+
+      -  the output will like this:
+
+         .. code:: makefile
+
+            clean: $(suite)         Will Clean the Environment of previous runs.
+            build: $(suite)         Will Build the GeoServer Docker Image for the Environment.
+            test: $(suite)      Will running the Suite test with teamengine.
+      - Choose which test to run by setting the Suite environment variable:
+
+        .. code:: SHELL
+
+           suite=wcs10
+
+        .. note::
+
+           Valid values for the Suite parameter are
+             * wcs10
+             * wcs11
+             * wfs10
+             * wfs11
+             * wms11
+             * wms13
+
+      - Choose which GeoServer war to test by setting the ``war_url`` environment variable inside the ``Makefile``, ex:
+
+        .. code:: C
+
+          war_url = "https://build.geoserver.org/geoserver/master/geoserver-master-latest-war.zip"
+
+      -  To clean the local environment.
+
+         .. code:: shell
+
+            make clean suite=<suite-name>
```

Comment:
the suite is an environment variable and the war-url is used inside the `Makefile` to build the GeoServer.

Reply:
"the suite is an environment variable" 

make it explicit in the doc then

"the war-url is used inside the Makefile to build the GeoServer."

can you make it a parameter of the makefile. How long would it take?

Reply:
> "the suite is an environment variable"
> 
> make it explicit in the doc then
> 
done.

> "the war-url is used inside the Makefile to build the GeoServer."
> 
> can you make it a parameter of the makefile. How long would it take?

if you want to change the war_url to another URL you can do it on the command...since is a variable in the `makefile`, just that the value declared there is the default value. ex.

``` shell
make build suite=wcs10 war_url="https://build.geoserver.org/geoserver/main/geoserver-main-latest-war.zip"
```

Reply:
solved